### PR TITLE
Ape-based generic subscription and authorization management CLI

### PR DESCRIPTION
**Pull request by KPrasch**
**Type of PR:**
Feature

**Required reviews:** 
2

**What this does:**
Introduces subscription-generic CLI that can be used for long-term management of multiple subscription contracts and authorizations for multiple adopters.  In contrast to PR #289 this PR implements both local and ape utilities.  As a result, it's necessary to install all the dependencies.

**Notes for reviewers:**
- Heavily inspired/adapted original work committed by @theref in PR #289
- Commits are co-authored by @theref and @derekpierre

### CODE REVIEW

**Review by theref — APPROVED** [review]

**Review by manumonti — APPROVED** [review]
This is going to be very useful 👌

**Review by vzotova — APPROVED** [review]

File: scripts/manage_subscription.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+@account_option()
+@domain_option
+@subscription_contract_option
+@encryptor_slots_option
+@click.option(
+    "--period",
+    default=0,
+    help="Subscription billing period number to pay for.",
+)
+def pay_subscription(account, domain, subscription_contract, encryptor_slots, period):
```
Comment (derekpierre) [inline scripts/manage_subscription.py:62]: If you wanted these commands to not use dashes in the CLI you could use the name option in the command() directive. For example, for this function you could do:
```python
@cli.command(name=paySubscription, cls=ConnectedProviderCommand)
...
```
Then the command would not use any dashes.

You can do the same for the `click.group()` i.e. `click.group(name="manageSubscription") for the `manage_subscription` CLI command group which derives its name from the file by default.

Up to you though - just giving options.

File: scripts/manage_subscription.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+def remove_encryptors(account, domain, ritual_id, access_controller, encryptors):
+    """Deauthorize encryptors from the access control contract for a ritual."""
+    transactor = Transactor(account=account)
+    access_controller = registry.get_contract(
+        contract_name=access_controller,
+        domain=domain
+    )
+    click.echo(
+        f"Removing {len(encryptors)} "
+        f"encryptors to the {access_controller} "
```
Comment (derekpierre) [inline scripts/manage_subscription.py:164]:
```suggestion
        f"encryptors from the {access_controller} "
```

File: scripts/manage_subscription.py (hunk 1 above)
Comment (KPrasch) [reply scripts/manage_subscription.py:62]: I gave this a try for the `click.group` but I did not have any success 🤷🏻 

```python
@click.group(name="manage-subscription")
def cli():
    """Subscription Management CLI"""
```

```python
(nucypher-contracts) k@Kierans-MacBook-Pro nucypher-contracts % ape run manage-subscription --help
Usage: ape run [OPTIONS] COMMAND [ARGS]...
Try 'ape run -h' for help.

Error: No such command 'manage-subscription'. Did you mean manage_subscription?
```

Comment (derekpierre) [reply scripts/manage_subscription.py:62]: Interesting. I guess the cli group name doesn't matter to ape when doing `ape run ...`.

Comment (derekpierre) [reply scripts/manage_subscription.py:62]: The "name" property for click.command should still work though (if desired).

Comment (KPrasch) [reply scripts/manage_subscription.py:62]: For sure, it works.  Personally I prefer the default naming behavior for CLI commands.

**Pull request merged**